.github/workflows/sonar-pr-from-fork-build.yml

```diff
@@ -0,0 +1,92 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: SonarBuild
+
+on: pull_request
+
+permissions:
+  contents: read
+
+concurrency:
+  group: sonar-pr-workflow-${{ github.event.pull_request.head.repo.full_name }}-${{ github.event.pull_request.head.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    if: github.repository == 'apache/ignite'
+    name: Build artifacts for Sonar Analysis
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: "refs/pull/${{ github.event.number }}/merge"
+          fetch-depth: 0
```

Comment:
> Number of commits to fetch. 0 indicates all history for all branches and tags.

I see that it actually fetches all branches and tags in this example - https://github.com/skorotkov/ignite/pull/61/checks. Can we reduce amount of data to fetch?

Reply:
This is used to correctly detect "new" code.

See more details at   https://community.sonarsource.com/t/git-fetch-depth-implications/75260

Reply:
May be we may get rid of this here since there is not actuar call to sonar here, but still should keep it in the sonar-pr-from-fork-scan.yml workflow.